impeller/aiks/paint_pass_delegate.cc

```diff
@@ -54,106 +53,4 @@ std::shared_ptr<FilterContents> PaintPassDelegate::WithImageFilter(
                                 Entity::RenderingMode::kSubpass);
 }
 
-/// OpacityPeepholePassDelegate
-/// ----------------------------------------------
-
-OpacityPeepholePassDelegate::OpacityPeepholePassDelegate(Paint paint)
-    : paint_(std::move(paint)) {}
-
-// |EntityPassDelgate|
-OpacityPeepholePassDelegate::~OpacityPeepholePassDelegate() = default;
-
-// |EntityPassDelgate|
-bool OpacityPeepholePassDelegate::CanElide() {
-  return paint_.blend_mode == BlendMode::kDestination;
-}
-
-// |EntityPassDelgate|
-bool OpacityPeepholePassDelegate::CanCollapseIntoParentPass(
-    EntityPass* entity_pass) {
-  // Passes with enforced bounds that clip the contents can not be safely
-  // collapsed.
-  if (entity_pass->GetBoundsLimitMightClipContent()) {
-    return false;
-  }
-
-  // OpacityPeepholePassDelegate will only get used if the pass's blend mode is
-  // SourceOver, so no need to check here.
-  if (paint_.color.alpha <= 0.0 || paint_.color.alpha >= 1.0 ||
```

Comment:
If the alpha is 0 then we should skip the pass entirely, so I'm not sure why it is saying that it "can't collapse into parent" here (i.e. return false).

Also, if the paint is opaque, then that isn't a reason not to collapse the pass...? It just means you have no partial opacity to distribute, but you can still collapse it and distribute the opacity of 1.0 (i.e. NOP on modifying the opacity).

Reply:
I think I was just being overly conservative at the time.